### PR TITLE
Added Auth

### PR DESCRIPTION
**Pull request by ayush65**
Signin
Signout
Logout

### CODE REVIEW

**Review by kotesh-arya — APPROVED** [review]
Everything looks well-defined and perfect for a merge 👍🏻

File: src/AllRoutes/AllRoutes.js [hunk 1]
```diff
@@ -4,12 +4,22 @@ import {
     Route
   } from "react-router-dom";
 import { Homepage } from '../pages/Homepage/Homepage';
+import { LandingPage } from '../pages/LandingPage/LandingPage';
+import { Logout } from '../pages/logout/Logout';
+import { SignIn } from '../pages/SignIn/SignIn';
+import { Signup } from '../pages/Signup/Signup';
```
Comment (mimansha-swarup) [inline src/AllRoutes/AllRoutes.js:10]: you can use barrel exports

File: src/Color/Color.css [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+    --primary-color : #041F1E ;
     --white: white ;
     --tertiary-color : #023E8A;
     --navbar-hover: #AED1E6;
     --navbar-shadow: rgba(50, 50, 93, 0.25);
     --navbar-shad: rgba(0, 0, 0, 0.3);
     --border-color: #777;
     --card-shadow: rgba(0, 0, 0, .2);
-    --background-color : #424242;
+    --background-color : #1E2D2F;
```
Comment (mimansha-swarup) [inline src/Color/Color.css:11]: good practice 👍

File: src/context/Auth-context.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { createContext, useContext, useReducer } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import {authReducer} from "../reducer/auth-reducer"
+
+
+const AuthContext = createContext(null);
+
+const AuthProvider = ({children}) => {
```
Comment (mimansha-swarup) [inline src/context/Auth-context.js:9]: nice Approach 👍

**Review by mimansha-swarup — APPROVED** [review]
clean and readable code
good practice in defining variables in CSS
variable and functions name is self-explanatory

**Review by Himanshu-Dhawale — APPROVED** [review]
Good practices followed.
Everything is self-explanatory.
Code is also clean.

File: src/AllRoutes/AllRoutes.js (hunk 1 above)
Comment (ayush65) [reply src/AllRoutes/AllRoutes.js]: done the changes what you have suggested . Thanks for reviewing

**Pull request merged**